File: 백준/Bronze/2875. 대회 or 인턴/대회 or 인턴.java
import java.util.*;

public class Main {
    public static void main(String args[]) {
      Scanner s = new Scanner(System.in);
      int n = s.nextInt(),m = s.nextInt(),k = s.nextInt();
      

      System.out.println(Math.min(Math.min(n/2,m),(n+m-k)/3));
    }
}

File: 백준/Bronze/2875. 대회 or 인턴/대회 or 인턴.cs
using System;

class Program
{
    static void Main() {
        int n,m,k;
        string[] input;
        input = Console.ReadLine().Split(' ');
        n = int.Parse(input[0]);
        m = int.Parse(input[1]);
        k = int.Parse(input[2]);

        Console.Write(Math.Min(Math.Min(n/2,m),(n+m-k)/3));
    }
}